client/my-sites/stats/features/modules/stats-locations/stats-locations.tsx

```diff
@@ -62,29 +66,64 @@ type SelectOptionType = {
 
 interface StatsModuleLocationsProps extends StatsDefaultModuleProps {
 	initialGeoMode?: string;
+	query: StatsQueryType;
+	context: {
+		query: {
+			[ key: string ]: string | undefined;
+		};
+	};
 }
 
 const StatsLocations: React.FC< StatsModuleLocationsProps > = ( {
 	initialGeoMode,
 	period,
 	query,
 	summaryUrl,
+	context,
+	summary,
 } ) => {
 	const translate = useTranslate();
 	const siteId = useSelector( getSelectedSiteId ) as number;
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
 	const statType = STAT_TYPE_COUNTRY_VIEWS;
 	const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
 	const supportUrl = isOdysseyStats
 		? `${ JETPACK_SUPPORT_URL_TRAFFIC }#views-by-locations`
 		: LOCATIONS_SUPPORT_URL;
+	const dispatch = useDispatch();
+	const [ selectedOption, setSelectedOption ] = useState( () => {
+		const urlGeoMode = initialGeoMode || context.query.geoMode;
+		return urlGeoMode && urlGeoMode in GEO_MODES ? urlGeoMode : OPTION_KEYS.COUNTRIES;
+	} );
+	const isStatsNavigationImprovementEnabled = config.isEnabled( 'stats/navigation-improvement' );
+
+	const [ countryFilter, setCountryFilter ] = useState< string | null >( null );
 
-	const urlGeoMode =
-		initialGeoMode || new URLSearchParams( window.location.search ).get( 'geoMode' );
-	const [ selectedOption, setSelectedOption ] = useState(
-		urlGeoMode && urlGeoMode in GEO_MODES ? urlGeoMode : OPTION_KEYS.COUNTRIES
+	const basePath = useMemo(
```

Comment:
Is this is only used in `useEffect` then I guess it could be move to that part.

Reply:
The changes related to location is moved to a new PR and this change is addressed in the new PR https://github.com/Automattic/wp-calypso/pull/103237, essentially I have removed all the code related to geoMode in favor of another PR - https://github.com/Automattic/wp-calypso/pull/103229